src/lib/common/port_monitor.c

```diff
@@ -0,0 +1,107 @@
+#include "port_monitor.h"
+#include "bm_config.h"
+#include "bm_os.h"
+#include "configuration.h"
+#include "device.h"
+#include "l2.h"
+#include "spotter.h"
+#include <inttypes.h>
+#include <string.h>
+
+#define port_monitor_time_ms_key "disableUnusedPortsTimeMs"
+#define port_monitor_time_ms_default (0U)
+#define port_monitor_log "port_monitor.log"
+
+static uint16_t ENABLED_PORT_MASK = 0;
+
+/*!
+  @brief Link change callback to set ports which have come online
+
+  @details Will set a masked value of the ports which have come online.
+
+  @param port port that has had a link change
+  @param state state of the port
+ */
+static void port_monitor_link_change(uint8_t port, bool state) {
```

Comment:
Please rename this arg (and in the comment) to indicate what representation we're talking about: `port_num`, `port_index`, or `port_mask`. Given the conversion to a mask after subtracting one I assume it's a `port_num`.